### PR TITLE
[Merged by Bors] - [profiler] Cache StringId

### PR DESCRIPTION
**Pull request by tunz**
boa_profiler was allocating new strings every time even when they are the same string, which leads to a crash when `start_event` is called too much. This change updates the profiler to use a hash map cache and avoid duplicate string allocations.

### CODE REVIEW

**Review by jedel1043 — APPROVED** [review]
Looks good! I would replace the `unwrap`s with `expect`s describing why unwrapping is fine though.

File: boa_profiler/src/lib.rs [hunk 1]
```diff
@@ -24,7 +24,7 @@
 //! [boa-web]: https://boa-dev.github.io/
 //! [boa-playground]: https://boa-dev.github.io/boa/playground/
 
-#![cfg_attr(not(test), forbid(clippy::unwrap_used))]
+#![cfg_attr(not(any(test, feature = "profiler")), forbid(clippy::unwrap_used))]
```
Comment (raskad) [inline boa_profiler/src/lib.rs:27]: Let's replace the `unwrap`s with `expects`s, then this change can also be reverted.

**Review by raskad — CHANGES_REQUESTED** [review]
Looks fine to me, but I'd say let's change the `unwrap`s like @jedel1043 suggested to stick to our normal lint rules.

**Review by HalidOdat — APPROVED** [review]
Looks good to me! :)

**Review by raskad — APPROVED** [review]

**Pull request closed without merging**